src/DcrdClient/DcrdHttpClient.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace DcrdClient
+{
+    /// <summary>
+    /// Http client to communicate with dcrd.
+    /// </summary>
+    public class DcrdHttpClient : IDcrdClient
+    {
+        private readonly string _apiUrl;
+        private readonly int _minConfirmations;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public DcrdHttpClient(DcrdHttpConfig config, IHttpClientFactory httpClientFactory)
+        {
+            _apiUrl = config.ApiUrl;
+            _httpClientFactory = httpClientFactory;
+            _minConfirmations = config.MinConfirmations;
+        }
+
+        private static DcrdRpcResponse<T> ParseResponse<T>(string responseBody)
+        {
+            try
+            {
+                return JsonConvert.DeserializeObject<DcrdRpcResponse<T>>(responseBody);
+            }
+            catch (Exception)
+            {
+                throw new DcrdException($"Failed to deserialize dcrd response: {responseBody}");
+            }
+        }
+
+        public async Task<DcrdRpcResponse<T>> PerformAsync<T>(string method, params object[] parameters)
+        {
+            using (var httpClient = _httpClientFactory.CreateClient())
```

Comment:
HttpClient is IDisposable, but it should not be disposed, until it is really necessary. See [this article](https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/) for additional information